src/lib/modules/blockchain_process/proxy.js

```diff
@@ -192,7 +183,18 @@ exports.serve = async (ipc, host, port, ws, origin) => {
       () => { resolve(server); }
     );
     if (ws) {
-      server.on('upgrade', proxy.upgrade);
+      server.on('upgrade', (req, socket, head) => {
+        const destroy = () => {
+          req.destroy();
+          socket.destroy();
+        };
+        const swallowECONNRESET = (_err) => {
+          setImmediate(destroy);
+        };
+        req.on('error', swallowECONNRESET);
+        socket.on('error', swallowECONNRESET);
```

Comment:
Looks like I made a mistake last night when rebasing and used an older commit... dangers of develping / testing on 3 different computers at the same time. I'm going to sort that out now.

Here's what that part will look like:
```js
if (ws) {
  server.on('upgrade', (msg, socket, head) => {
    const swallowError = (err) => {
      console.error(`Proxy: Network error '${err.message}'`);
    };
    socket.on('error', swallowError);
    proxy.upgrade(msg, socket, head);
  });
}
```

I checked yesterday and here's what [the docs](https://nodejs.org/dist/latest-v10.x/docs/api/net.html#net_event_error_1) say about `net.Socket`'s `'errror'` event:
> Emitted when an error occurs. The `'close'` event will be called directly following this event.

Since the socket will be closed, no need to destroy it manually and there's not much more we can / should do at that point other than report with `console.error()`.

Also, the variable I had named `req` previously is actually an [`http.IncomingMessage`](https://nodejs.org/dist/latest-v10.x/docs/api/http.html#http_class_http_incomingmessage) instance. It lacks an `'error'` event and its `destroy()` method just destroys the underlying socket (`msg.socket`, alias `msg.connection`), which is identity with the `socket` variable, so we don't need to do anything with it.